docs/core/Lifecycle.md

```diff
@@ -0,0 +1,102 @@
+# Carbon Lifecycle
+
+For a consistent development experience Carbon uses the same underlying interface for all graph types. All Carbon graphs are separated into 2 parts: `Construct` & `GraphContent`.
+
+-   [Carbon Lifecycle](#carbon-lifecycle)
+    -   [Construct](#construct)
+        -   [`loadContent(input)`](#loadcontentinput)
+        -   [`unloadContent(input)`](#unloadcontentinput)
+        -   [`destroy`](#destroy)
+    -   [Important information](#important-information)
+    -   [GraphContent](#graphcontent)
+
+## Construct
+
+`Construct` is the base for `Graph` and other types of graph that serves as a container for the datasets (content). Usually this takes care of one or more of the following:
+
+-   X Axis
+-   Y Axis
+-   Vertical Grid
+-   Horizontal Grid
+-   X Axis labels
+-   Y Axis labels
+-   Legend
+
+Construct exposes following functions to handle the graph or its contents:
+
+-   `generate` - Called internally when instantiating the graph
+-   `loadContent` - Load a dataset in the graph
+-   `unloadContent` - Remove a dataset from the graph
+-   `resize` - Resizes the graph based on window resize event
+-   `destroy` - Destroys the graph
+
+### `loadContent(input)`
+
+For Line, Paired Result and Bar graphs:
+
+```javascript
+var data = {
+    key: "uid_1",
+    label: {
+        display: "Data Label 1"
+    },
+    values: [
+        {
+            x: "2016-03-01T12:00:00Z",
+            y: 14
+        },
+        {
+            x: "2016-04-10T12:00:00Z",
+            y: 1
+        },
+        {
+            x: "2016-11-01T12:00:00Z",
+            y: 18
+        }
+    ]
+};
+var lineGraph = Carbon.api.graph(root);
```

Comment:
I think it would be nice to say what the `root` here is. Just to make sure there wouldn't be any gaps